### PR TITLE
Add support for an `inherits_from` directive

### PR DESCRIPTION
**Pull request by michaelherold**
Borrowing from RuboCop, the idea of inheriting a configuration from
another file is useful because it allows you to have a central
configuration for your team to use across multiple projects. It's also
a necessary feature to make an `--auto-gen-config` command useful by
giving you a configuration that you can inherit from that is generated
from an existing code base.

### CODE REVIEW

**Comment by michaelherold** [comment]
🤔 I wrote this as "inherits_from" because I thought it read well, but then I noticed that RuboCop has their functionality like this as "inherit_from". I wonder if I should switch it over to match or keep the version that reads better?

File: spec/haml_lint/configuration_loader_spec.rb [hunk 1]
```diff
@@ -109,6 +109,146 @@
...
+          'skip_frontmatter: true',
+          'linters:',
+          '  AltText:',
+          '    enabled: true'
+        ].join("\n")
+      end
+      let(:other_config) { 'other_haml-lint.yml' }
+
+      before do
+        File.open(file_name, 'w') { |f| f.write(config_file) }
```
Comment (trotzig) [inline spec/haml_lint/configuration_loader_spec.rb]: I might be missing something here, but I don't see `file_name` being defined anywhere. Also, don't you need to do some cleanup after the test is done? (bare with me, I haven't written much code in this project).

File: spec/haml_lint/configuration_loader_spec.rb [hunk 2]
```diff
@@ -109,6 +109,146 @@
...
+              'AltText' => { 'enabled' => true },
+              'Indentation' => { 'enabled' => false }
+            }
+          )
+
+          subject.should ==
+            described_class.default_configuration.merge(custom_config)
+        end
+
+        context 'when the file has a circular dependency on another' do
```
Comment (trotzig) [inline spec/haml_lint/configuration_loader_spec.rb]: Thanks for taking care of this case!

File: spec/haml_lint/configuration_loader_spec.rb (hunk 1 above)
Comment (trotzig) [inline spec/haml_lint/configuration_loader_spec.rb]: Ah, `file_name` wasn't added by you. And I take it the `around` block up top takes care of cleaning things up.

Comment (michaelherold) [inline spec/haml_lint/configuration_loader_spec.rb]: Yep, the `around` creates an isolated environment for creating the temp files. The tests for this class are almost to the point where they are hard to follow.

**Comment by michaelherold** [comment]
Good idea on allowing both `inherit_from` and `inherits_from`. I'll add that tonight!

**Comment by michaelherold** [comment]
I added `inherit_from` as an alias that combines with `inherits_from`. What do you think?

**Comment by trotzig** [comment]
Looks good!
